core-test/src/test/scala/org/bitcoins/core/psbt/PSBTTest.scala

```diff
@@ -233,6 +239,39 @@ class PSBTTest extends BitcoinSAsyncTest {
                                             getDummySigners(size = 1)))
   }
 
+  it must "correctly construct and sign a PSBT" in {
+    forAllAsync(PSBTGenerators.psbtToBeSigned) { psbtWithBuilderF =>
+      psbtWithBuilderF.flatMap {
+        case (psbtNoSigs, utxos) =>
+          val infos = utxos.toVector.zipWithIndex.map {
+            case (utxo: BitcoinUTXOSpendingInfoFull, index) =>
+              (index, utxo)
+          }
+          val signedPSBTF = infos.foldLeft(Future.successful(psbtNoSigs)) {
+            case (unsignedPSBTF, (index, info)) =>
+              unsignedPSBTF.flatMap { unsignedPSBT =>
+                info.toSingles.foldLeft(Future.successful(unsignedPSBT)) {
+                  (psbtToSignF, singleInfo) =>
+                    psbtToSignF.flatMap(
+                      _.sign(index,
+                             singleInfo.signer,
+                             singleInfo.conditionalPath,
+                             test = Some(info)))
+                }
+              }
+          }
+          signedPSBTF.map { signedPSBT =>
+            val finalizedPsbtT = signedPSBT.finalizePSBT
+            finalizedPsbtT match {
+              case Success(finalizedPsbt) =>
+                assert(finalizedPsbt.extractTransactionAndValidate.isSuccess)
+              case Failure(exception) => throw exception
```

Comment:
use `fail` method instead which will wrap the exception in a nice and descriptive `TestFailedException` (which uses reflection and stuff)